lib/rubocop.rb

```diff
@@ -750,6 +750,11 @@
 require_relative 'rubocop/config_validator'
 require_relative 'rubocop/feature_loader'
 require_relative 'rubocop/lockfile'
+require_relative 'rubocop/lsp'
+require_relative 'rubocop/lsp/logger'
+require_relative 'rubocop/lsp/routes'
+require_relative 'rubocop/lsp/runtime'
+require_relative 'rubocop/lsp/severity'
```

Comment:
These `require_relative` codes should only be necessary when running `rubocop --lsp` command. Is it really true that requiring these LSP-related codes globally in `lib/rubocop.rb` speeds up the regular `rubocop` command as well?

Reply:
Indeed, the only one necessary is the one for `RuboCop::LSP.enabled?` and similar that some cops are using. These don't really matter since they are such small files, the real gain is from not requiring `language-server_protocol` every time.

These requires where only needed for some other approach I tried and I had them mixed now. Reran the benchmark with the revised change and its basically the same still.

Reply:
For reference, here is the flamegraph:

![grafik](https://github.com/user-attachments/assets/0eb4bc8a-d648-4f1d-97e6-6fd10e69fa71)

Most of the small ones to the left are cop requires, and the one I'm hovering over is the gem (notice `module:LanguageServer` a bit on top)